src/main/java/com/maddyhome/idea/vim/extension/nerdtree/Mappings.kt

```diff
@@ -0,0 +1,165 @@
+
+  fun register(variable: String, defaultMapping: String, action: Action) {
+    val variableValue = VimPlugin.getVariableService().getGlobalVariableValue(variable)
+    val mapping = if (variableValue is VimString) {
+      variableValue.value
+    } else {
+      defaultMapping
+    }
+    register(mapping, action)
+  }
+
+  fun register(mapping: String, action: Action) {
+    trie.add(mapping, action)
+    _keyStrokes.addAll(injector.parser.parseKeys(mapping))
+  }
+
+  /**
+   * Defines the actual behavior of actions in NERDTree
+   */
+  class Action(val action: (AnActionEvent, Tree) -> Unit) {
+    companion object {
+      private fun executeAction(event: AnActionEvent, id: String) =
+        NerdTree.Util.callAction(null, id, event.dataContext.vim)
+
+      /**
+       * Creates an Action that executes a series of IntelliJ actions identified by their IDs.
+       *
+       * @param ids A variable number of strings representing the IDs of the actions to execute.
+       * @return An Action that runs the specified actions when triggered.
+       */
+      fun ij(vararg ids: String): Action = Action { event, _ -> ids.forEach { id -> executeAction(event, id) } }
+    }
+  }
+
+  /**
+   * Registers navigation-related mappings
+   * <pre><code>
+   * Default~
+   * Key      Description                                                             Map settings
+   *
+   * O........Recursively open the selected directory..................*NERDTreeMapOpenRecursively*
+   * x........Close the current nodes parent..................................*NERDTreeMapCloseDir*
+   * X........Recursively close all children of the current node.........*NERDTreeMapCloseChildren*
+   *
+   * P........Jump to the root node...........................................*NERDTreeMapJumpRoot*
+   * p........Jump to current nodes parent..................................*NERDTreeMapJumpParent*
+   * K........Jump up inside directories at the current tree depth......*NERDTreeMapJumpFirstChild*
+   * J........Jump down inside directories at the current tree depth.....*NERDTreeMapJumpLastChild*
+   * <C-J>....Jump down to next sibling of the current directory.......*NERDTreeMapJumpNextSibling*
+   * <C-K>....Jump up to previous sibling of the current directory.....*NERDTreeMapJumpPrevSibling*
+   * </code></pre>
+   */
+  fun registerNavigationMappings() {
+    // TODO support going [count] lines upward/downward or to line [count]
+    register("k", Action.ij("Tree-selectPrevious"))
+    register("j", Action.ij("Tree-selectNext"))
+    register("G", Action.ij("Tree-selectLast"))
+    register("gg", Action.ij("Tree-selectFirst"))
+
+    // FIXME lazy loaded tree nodes are not expanded
+    register("NERDTreeMapOpenRecursively", "O", Action.ij("FullyExpandTreeNode"))
+    // This action respects `ide.tree.collapse.recursively`. We may have to ask the user to disable it
+    register("NERDTreeMapCloseDir", "x", Action.ij("Tree-selectParentNoCollapse", "CollapseTreeNode"))
```

Comment:
Do we need to **temporarily** disable `ide.tree.collapse.recursively` (by `AdvancedSettings.setBoolean`) to make the action behavior identical to the one in Vim NERDTree? Or prompt the user to do so manually?

Reply:
Oh oh, after I answered to your comment, I've noticed this ticket
[VIM-3437](https://youtrack.jetbrains.com/issue/VIM-3437) NERDtree: "Recursively close all children of the current node" does not work

Well, in this case, since the NERDTree documentation explicitly states that it recursively closes the nodes, then we have to obey and also do it recursively.